### PR TITLE
update lints

### PR DESCRIPTION
**Pull request by kevmoo**
- Update lints, add and fix lints, lengthen description
- Also add example – and cleanup example in readme

### CODE REVIEW

**Comment by kevmoo** [comment]
Feel free to squash/merge and delete the branch if you're happy

File: pubspec.yaml [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 repository: https://github.com/dart-lang/test_descriptor
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: '>=2.17.0 <3.0.0'
 
 dependencies:
   async: ^2.5.0
   collection: ^1.15.0
-  matcher: ^0.12.10
```
Comment (kevmoo) [inline pubspec.yaml:12]: not used!

File: pubspec.yaml [hunk 2]
```diff
@@ -1,19 +1,17 @@
 name: test_descriptor
-version: 2.0.1
-description: An API for defining and verifying directory structures.
+version: 2.0.2-dev
+description: An API for defining and verifying files and directory structures.
```
Comment (kevmoo) [inline pubspec.yaml:3]: pushing for ~60 chars

File: pubspec.yaml [hunk 3]
```diff
@@ -1,19 +1,17 @@
 name: test_descriptor
-version: 2.0.1
-description: An API for defining and verifying directory structures.
+version: 2.0.2-dev
```
Comment (devoncarew) [inline pubspec.yaml:2]: Please add a matching version in the changelog

**Review by devoncarew — APPROVED** [review]
lgtm modulo adding a matching version in the changelog (to match the rev to `2.0.2-dev`)

**Pull request merged**